### ISSUE TITLE
Allow runtime loading of unicode database used

### ISSUE THREAD

**Issue by horazont** [byllyfish/precis_i18n#8]
In contrast to Stringprep, PRECIS is not bound to a fixed version of the unicode database. How the issues resulting from using different Unicode database versions in a multi-party system are resolved are left to the protocols making use of PRECIS.

One obvious solution is to pin the Unicode database, another solution is to be extremely agile (i.e. update fast), yet another is to negotiate the version used.

In any case, it is required that users of PRECIS have precise control over the unicode database version used.

Thus, it would be immensely useful if the PRECIS library was able to load a Unicode Character Database from a file and use that at runtime instead of the Python-provided `unicodedata`.

**Comment by byllyfish** [comment]
Currently, the `unicodedata` version is tied to the version of the Python interpreter. For Python 3.7, it's Unicode 11.0.  Python 3.8 will be Unicode 12.1.

The PRECIS library could support an alternate `unicodedata` library, e.g. `unicodedata2`. This would give you better control over the version of Unicode.

I envision it would be used like this. First, pip install a specific version:

```
pip install unicodedata2==10.0.0
```

Then, use the specific version by passing it as a keyword argument to get_profile:

```python
import unicodedata2
from precis_i18n import get_profile

username = get_profile('UsernameCaseMapped', unicodedata=unicodedata2)
```

Will that help?

**Comment by horazont** [comment]
I’m not sure. It would be better to be able to select the Unicode version at runtime, and potentially have different Unicode versions loaded at the same time, e.g. when in a situation where the unicode version used is negotiated at runtime in the protocol.

I imagine it to be similar to how the `random` package works. You can either use `random.getrandint` and it’ll use the built-in PRNG, or you can instantiate a separate PRNG (possibly even SystemRandom) and use the same methods on that.

Being able to use ``precis_i18n`` with alternate unicodedata implementations would be a good start in that direction though.

**Comment by byllyfish** [comment]
I agree that selecting the Unicode version at runtime would be better. My plan is to first add the unicodedata override. Then, I will take a look at the possibility of a custom unicodedata implementation that loads version data from a file for the subset of features needed for PRECIS.
